src/lib/modules/Document/DocumentSubjectSearch.js

```diff
@@ -0,0 +1,148 @@
+import { documentApi } from '@api/documents';
+import { Media } from '@components/Media';
+import {
+  setReactSearchKitDefaultSortingOnEmptyQueryString,
+  setReactSearchKitInitialQueryState,
+  setReactSearchKitUrlHandler,
+} from '@config';
+import history from '@history';
+import { SearchControls } from '@modules/SearchControls/SearchControls';
+import { SearchControlsOverridesMap } from '@modules/SearchControls/SearchControlsOverrides';
+import SearchFooter from '@modules/SearchControls/SearchFooter';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { OverridableContext } from 'react-overridable';
+import {
+  EmptyResults,
+  Error,
+  InvenioSearchApi,
+  ReactSearchKit,
+  ResultsLoader,
+  ResultsGrid,
+} from 'react-searchkit';
+import { Divider, Container, Ref, Sticky } from 'semantic-ui-react';
+import _isEmpty from 'lodash/isEmpty';
+import { InvenioRequestSerializer } from 'react-searchkit';
+import { DocumentSubjectGrid } from './DocumentSubjectLayout';
+import { Error as IlsError } from '@components/Error';
+import { SearchControlsMobile } from '@modules/SearchControls/SearchControlsMobile';
+
+const getSubjectsQuery = (documentMetadata) => {
+  const subjectSchemes = documentMetadata.subjects.map((subject) => {
```

Comment:
this part looks like it could fit into QueryBuilder class: https://github.com/inveniosoftware/react-invenio-app-ils/blob/master/src/lib/api/documents/document.js#L56